pandas/core/indexing.py

```diff
@@ -2063,7 +2063,17 @@ def _convert_key(self, key, is_setter: bool = False):
 
         return key
 
+    @property
+    def _axes_are_unique(self) -> bool:
+        # Only relevant for self.ndim == 2
+        assert self.ndim == 2
+        return self.obj.index.is_unique and self.obj.columns.is_unique
+
     def __getitem__(self, key):
+        if self.ndim == 2 and not self._axes_are_unique:
+            # GH#33041 fall back to .loc
+            return self.obj.loc[key]
```

Comment:
One problem with this, though, is that it allows any kind of indexer for this case (not only scalar ones), and thus relaxing the requirements for `.at`

Reply:
Updated to address this issue